playground/backend/internal/fs_tool/fs_test.go

```diff
@@ -30,132 +28,108 @@ import (
 const (
 	sourceDir       = "sourceDir"
 	destinationDir  = "destinationDir"
+	testFileMode    = 0755
 	pipelinesFolder = "executable_files"
 )
 
-func TestMain(m *testing.M) {
-	err := setupPreparedFiles()
-	if err != nil {
-		logger.Fatal(err)
-	}
-	defer teardown()
-	m.Run()
-}
-
-func setupPreparedFiles() error {
-	err := os.Mkdir(sourceDir, 0755)
+func prepareFiles() error {
+	err := os.Mkdir(sourceDir, testFileMode)
 	if err != nil {
 		return err
 	}
-	err = os.Mkdir(destinationDir, 0755)
+	err = os.Mkdir(destinationDir, testFileMode)
 	if err != nil {
 		return err
 	}
 	filePath := filepath.Join(sourceDir, "file.txt")
 	_, err = os.Create(filePath)
-	if err != nil {
-		return err
-	}
-	return nil
+	return err
 }
 
-func teardown() {
+func teardownFiles() error {
 	err := os.RemoveAll(sourceDir)
 	if err != nil {
-		logger.Fatal(err)
-	}
-	err = os.RemoveAll(destinationDir)
-	if err != nil {
-		logger.Fatal(err)
-	}
-	err = os.RemoveAll(pipelinesFolder)
-	if err != nil {
-		logger.Fatal(err)
+		return err
 	}
+	return os.RemoveAll(destinationDir)
 }
 
-func TestLifeCycle_CreateExecutableFile(t *testing.T) {
-	pipelineId := uuid.New()
-	baseFileFolder := fmt.Sprintf("%s/%s", pipelinesFolder, pipelineId)
-	srcFileFolder := baseFileFolder + "/src"
-	binFileFolder := baseFileFolder + "/bin"
+func prepareFolders(baseFileFolder string) error {
+	srcFileFolder := filepath.Join(baseFileFolder, "src")
+
+	return os.MkdirAll(srcFileFolder, testFileMode)
+}
+
+func teardownFolders(baseFileFolder string) error {
```

Comment:
Why have you divided teardown to two methods?

Reply:
Ditto as the previous answer.